src/OpenTelemetry.Exporter.Zipkin/Implementation/ZipkinExporterEventSource.cs

```diff
@@ -0,0 +1,66 @@
+﻿// <copyright file="ZipkinExporterEventSource.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Diagnostics.Tracing;
+using System.Globalization;
+using System.Threading;
+
+namespace OpenTelemetry.Exporter.Zipkin.Implementation
+{
+    /// <summary>
+    /// EventSource events emitted from the project.
+    /// </summary>
+    [EventSource(Name = "OpenTelemetry-Exporter-Zipkin")]
+    internal class ZipkinExporterEventSource : EventSource
+    {
+        public static ZipkinExporterEventSource Log = new ZipkinExporterEventSource();
+
+        [NonEvent]
+        public void FailedExport(Exception ex)
+        {
+            if (this.IsEnabled(EventLevel.Error, (EventKeywords)(-1)))
+            {
+                this.FailedExport(ToInvariantString(ex));
+            }
+        }
+
+        [Event(1, Message = "Failed to export activities: '{0}'", Level = EventLevel.Error)]
+        public void FailedExport(string exception)
```

Comment:
Let's change to private, so users only use the public methods which will validate if the log is enabled or not.

Reply:
The class is internal so its not public to end users.
Its okay to use this method directly, if the exception string is readily available (for some other purposes). The protection with non-event, is to avoid the conversion costs, unless listener is enabled.